client/signup/steps/site-picker/site-picker-submit.jsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { PLAN_FREE } from 'lib/plans/constants';
+import { getSite } from 'state/sites/selectors';
+import SignupActions from 'lib/signup/actions';
+
+class SitePickerSubmit extends Component {
+	componentWillMount() {
```

Comment:
Oh interesting, we could also use the `componentDidUpdate` lifecycle, if you don't want to guard in the parent component.